File: C/55.JumpGame.c
bool canJump(int* nums, int numsSize){
    int farest = 0;
    for (int i=0; i<numsSize-1; i++) {
        if (farest < i || farest >= numsSize-1) break;
        if (farest < i+nums[i]) farest = i+nums[i];
    }
    return farest >= (numsSize-1);
}

File: Go/55.JumpGame.go
func canJump(nums []int) bool {
    i := 0
    l := len(nums)
    for i < (l - 1) { // stand before the last index
        jump := nums[i]
        if jump == 0 {
            return false
        }
        cur := i+jump
        farest := cur
        if farest >= (l - 1) { // beyond the last
            return true
        }
        //fmt.Println("for", i+1, i+jump)
        for j:=i+1; j < cur ; j++ {
            if j + nums[j] > farest {
                farest = j + nums[j]
            }
        }
        i = cur
        if i + nums[i] < farest {
            nums[i] = farest - i
        }
    }
    return true
}
